src/loop/parser_intf.ml

```diff
@@ -33,11 +33,23 @@ module type S = sig
   *)
 
   val parse_logic :
+    ?preludes:Logic.language State.file list ->
     Dolmen.Std.Statement.t list -> state -> Logic.language State.file ->
     state * (state -> state * Dolmen.Std.Statement.t option)
```

Comment:
Since we're breaking interfaces for the upcoming release, I wonder if it wouldn't be simpler to change this function to instead take as main argument a list of files (rather than a single file).
Additionally, I'm not sure that the list of statements as prelude is really useful (and in any case, it's trivial for the user/caller to manually prepend a chosen list of statements), so we might remove it, considering how we've seen that it's not simple to use it correctly.

Reply:
I think from the user's point of view separating preludes and main file makes for a cleaner interface, even if there is no difference implementation-wise.

Agreed with removing the list of statements as prelude, I did not touch it because I did not need to change the interface of `parse_logic` otherwise (sure adding `?preludes` is technically a breaking change but probably won't actually break code in practice).

Reply:
I kept `?preludes:Logic.language State.file list -> state -> Logic.language State.file -> ...` so as not to lose the ability to do what you describe in the future.

While I am at it — should I also change `parse_response`, at least for consistency?

Reply:
One difference is that only the main file is allowed to be `Stdin` actually

Reply:
> While I am at it — should I also change `parse_response`, at least for consistency?

I have now read the code of `parse_response` and now think that it is fine to leave it as is (: